evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,114 @@
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+        arima = import_or_raise("statsmodels.tsa.arima_model", error_msg=p_error_msg)
+        arima.ARIMA(endog=np.zeros(p + d + q + 1), **parameters)  # to validate parameters
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_seed=random_seed)
+
+    def get_dates_fit(self, X, y):
+        date_col = None
+
+        if isinstance(y.index, pd.DatetimeIndex):
+            date_col = y.index
+        if X is not None:
+            if self.date_column in X.columns:
+                date_col = X.pop(self.date_column)
+                X.index = y.index
+            elif isinstance(X.index, pd.DatetimeIndex):
+                date_col = X.index
+            X.index = y.index
+
+        if date_col is None:
+            msg = "ARIMA regressor requires input data X to have a datetime column specified by the 'date_column' parameter. " \
+                  "If not it will look for the datetime column in the index of X or y."
+            raise ValueError(msg)
+        return date_col
+
+    def get_dates_predict(self, X, y):
+        date_col = None
+
+        if y is not None:
+            if isinstance(y.index, pd.DatetimeIndex):
+                date_col = y.index
+        if X is not None:
+            if self.date_column in X.columns:
+                date_col = X.pop(self.date_column)
+            elif isinstance(X.index, pd.DatetimeIndex):
+                date_col = X.index
+
+        if date_col is None:
+            msg = "ARIMA regressor requires input data X to have a datetime column specified by the 'date_column' parameter. " \
+                  "If not it will look for the datetime column in the index of X."
+            raise ValueError(msg)
+        return date_col
+
+    def fit(self, X, y=None):
+        if y is None:
+            raise ValueError('ARIMA Regressor requires y as input.')
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+        arima = import_or_raise("statsmodels.tsa.arima_model", error_msg=p_error_msg)
+
+        X, y = self._manage_woodwork(X, y)
+        dates = self.get_dates_fit(X, y)
+        if X is not None:
+            arima_with_data = arima.ARIMA(endog=y, exog=X, dates=dates, **self.parameters)
+        else:
+            arima_with_data = arima.ARIMA(endog=y, dates=dates, **self.parameters)
+
+        self._component_obj = arima_with_data
+        self._component_obj.fit(solver='nm', disp=0)
+        return self
+
+    def predict(self, X, y=None):
+        X, y = self._manage_woodwork(X, y)
+        dates = self.get_dates_predict(X, y)
+        start = dates.min()
+        end = dates.max()
+        params = self.parameters['order']
+        if X is not None:
+            y_pred = self._component_obj.predict(params=params, start=start, end=end, exog=X)
```

Comment:
In the statsmodel docs they use the ARIMA object slightly differently. Basically they call `fit` to get a "results object" and then call `predict` from the "results object" and not the original ARIMA object. [Example](https://www.statsmodels.org/devel/examples/notebooks/generated/statespace_forecasting.html#:~:text=They%20are%20predict%20and%20get_prediction,results%20(similar%20to%20get_forecast%20).).

I think we should follow that approach unless there's a good reason not too!